### PR TITLE
fix(queries): align queries cross different platforms

### PR DESCRIPTION
**Pull request by roi-orca**
align queries cross different platforms

References:
https://github.com/Checkmarx/kics/pull/5460
https://github.com/Checkmarx/kics/pull/5446

**Proposed Changes**
align queries cross different platforms

Queries:

1. API Gateway without WAF
2. CloudTrail Log File Validation Disabled
3. CloudTrail Log Files Not Encrypted With CMK
4. DB Instance Publicly Accessible
5. Hardcoded AWS Access Key In Lambda
6. IAM Password Without Number
7. S3 Bucket Logging Disabled
8. SQL Analysis Services Port 2383 (TCP) Is Publicly Accessible
9. Stack Notifications Disabled
10. Azure Container Registry With No Locks
11. Public Storage Account
12. Redis Cache Allows Non SSL Connections
13. COS Node Image Not Used
14. Google Compute Subnetwork with Private Google Access Disabled
15. High Google KMS Crypto Key Rotation Period
16. High KMS Rotation Period
17. Serial Ports Are Enabled For VM Instances
18. SSH Access Is Not Restricted
19. Using Default Service Account
20. Cloudfront Logging Disabled
21. Cloudfront Without WAF
22. CodeBuild Not Encrypted
23. DB Security Group With Public Scope
24. ECS Service Admin Role is Present
25. User Data Contains Encoded Private Key
26. Authentication Without MFA
27. CA Certificate Identifier Is Outdated
28. ElasticSearch Without Slow Logs
29. Hardcoded AWS Access Key
30. SQS with SSE disabled
31. AKS Network Policy Misconfigured
32. CosmosDB Account IP Range Filter Not Set
33. Role Definition Allows Custom Role Creation
34. Unrestricted SQL Server Access

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/terraform/gcp/vm_serial_ports_are_enabled_for_vm_instances/metadata.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "queryName": "Serial Ports Are Enabled For VM Instances",
   "severity": "MEDIUM",
   "category": "Insecure Configurations",
-  "descriptionText": "Check if VM instance enables serial ports",
+  "descriptionText": "Check if serial ports are enabled in Google Compute Engine VM instances",
```
Comment (rafaela-soares) [inline assets/queries/terraform/gcp/vm_serial_ports_are_enabled_for_vm_instances/metadata.json:6]: For all "Serial Ports Are Enabled For VM Instances" queries:

```suggestion
  "descriptionText": "Google Compute Engine VM instances should not enable serial ports",
```

**Review by rafaela-soares — CHANGES_REQUESTED** [review]
Hi, @roi-orca!

Thank you so much for continuing to improve the KICS queries metadata! 🚀 

I suggest the following:

File: assets/queries/ansible/aws/hardcoded_aws_access_key_in_lambda/metadata.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "queryName": "Hardcoded AWS Access Key In Lambda",
   "severity": "MEDIUM",
   "category": "Secret Management",
-  "descriptionText": "Lambda access key should not be in plaintext.",
+  "descriptionText": "Lambda hardcoded AWS access/secret keys",
```
[the remark this answers is not in the document]
Comment (roi-orca) [reply assets/queries/ansible/aws/hardcoded_aws_access_key_in_lambda/metadata.json]: ok

File: assets/queries/ansible/aws/stack_notifications_disabled/metadata.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "queryName": "Stack Notifications Disabled",
   "severity": "MEDIUM",
   "category": "Observability",
-  "descriptionText": "AWS CloudFormation should have stack notifications enabled",
+  "descriptionText": "Enable AWS CloudFormation Stack Notifications",
```
[the remark this answers is not in the document]
Comment (roi-orca) [reply assets/queries/ansible/aws/stack_notifications_disabled/metadata.json]: ok and for all

File: assets/queries/ansible/gcp/ssh_access_is_not_restricted/metadata.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
   "queryName": "SSH Access Is Not Restricted",
   "severity": "MEDIUM",
   "category": "Networking and Firewall",
-  "descriptionText": "Check if Google Firewall allows SSH access (port 22) from the Internet (public CIDR block).",
+  "descriptionText": "Check if Google Firewall allows SSH access (port 22) from the Internet (public CIDR block)",
```
[the remark this answers is not in the document]
Comment (roi-orca) [reply assets/queries/ansible/gcp/ssh_access_is_not_restricted/metadata.json]: ok for all

File: assets/queries/cloudFormation/aws/cloudfront_logging_disabled/metadata.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "id": "de77cd9f-0e8b-46cc-b4a4-b6b436838642",
-  "queryName": "CloudFront Logging Disabled",
+  "queryName": "Cloudfront Logging Disabled",
```
[the remark this answers is not in the document]
Comment (roi-orca) [reply assets/queries/cloudFormation/aws/cloudfront_logging_disabled/metadata.json]: ok

File: assets/queries/cloudFormation/aws/ecs_service_admin_role_is_present/metadata.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "id": "01986452-bdd8-4aaa-b5df-d6bf61d616ff",
-  "queryName": "ECS Service Admin Role Is Present",
+  "queryName": "ECS Service Admin Role is Present",
```
[the remark this answers is not in the document]
Comment (roi-orca) [reply assets/queries/cloudFormation/aws/ecs_service_admin_role_is_present/metadata.json:3]: There are a lot of queries that some are uppercased and some not, I did the alignment on those that are the same query name for now.. we can next pr decide if we want one or another all over the files.. that sound reasonable?

File: assets/queries/cloudFormation/aws/user_data_contains_encoded_private_key/metadata.json [hunk 7]
```diff
@@ -3,7 +3,7 @@
   "queryName": "User Data Contains Encoded Private Key",
   "severity": "HIGH",
   "category": "Encryption",
-  "descriptionText": "User Data Base64 contains an encoded RSA Private Key",
+  "descriptionText": "User Data contains an encoded RSA Private Key",
```
[the remark this answers is not in the document]
Comment (roi-orca) [reply assets/queries/cloudFormation/aws/user_data_contains_encoded_private_key/metadata.json]: ok to all

File: assets/queries/terraform/aws/hardcoded_aws_access_key/metadata.json [hunk 8]
```diff
@@ -3,7 +3,7 @@
   "queryName": "Hardcoded AWS Access Key",
   "severity": "LOW",
   "category": "Secret Management",
-  "descriptionText": "Hard-coded AWS access key / secret key exists in EC2 user data",
+  "descriptionText": "Check if the user data in the EC2 instance has the access key hardcoded",
```
[the remark this answers is not in the document]
Comment (roi-orca) [reply assets/queries/terraform/aws/hardcoded_aws_access_key/metadata.json]: ok to all

File: assets/queries/terraform/aws/sqs_with_sse_disabled/metadata.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "id": "6e8849c1-3aa7-40e3-9063-b85ee300f29f",
-  "queryName": "SQS With SSE Disabled",
+  "queryName": "SQS with SSE disabled",
```
[the remark this answers is not in the document]
Comment (roi-orca) [reply assets/queries/terraform/aws/sqs_with_sse_disabled/metadata.json]: ok to all

File: assets/queries/terraform/azure/azure_container_registry_with_no_locks/metadata.json [hunk 10]
```diff
@@ -3,7 +3,7 @@
   "queryName": "Azure Container Registry With No Locks",
   "severity": "HIGH",
   "category": "Insecure Configurations",
-  "descriptionText": "Azurerm Container Registry Must Contain Associated Locks ",
+  "descriptionText": "Azurerm Container Registry Should Contain Associated Locks",
```
[the remark this answers is not in the document]
Comment (roi-orca) [reply assets/queries/terraform/azure/azure_container_registry_with_no_locks/metadata.json]: ok

File: assets/queries/terraform/gcp/vm_serial_ports_are_enabled_for_vm_instances/metadata.json (hunk 1 above)
Comment (roi-orca) [reply assets/queries/terraform/gcp/vm_serial_ports_are_enabled_for_vm_instances/metadata.json]: ok to all

File: assets/queries/cloudFormation/aws/ecs_service_admin_role_is_present/metadata.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (rafaela-soares) [reply assets/queries/cloudFormation/aws/ecs_service_admin_role_is_present/metadata.json:3]: Yes, @roi-orca! Thank you so much for all the work! 🚀

**Review by rafaela-soares — COMMENTED** [review]
LGTM 🚀 

I will only wait for our security to confirm the severity and category alignment.

**Comment by roi-orca** [comment]
@rafaela-soares any update on this?

**Comment by rafaela-soares** [comment]
> @rafaela-soares any update on this?

Hello, @roi-orca! The PR is already being reviewed. Sorry for the delay. We will update you, as soon as possible 😊

File: assets/queries/ansible/azure/public_storage_account/metadata.json [hunk 11]
```diff
@@ -3,7 +3,7 @@
   "queryName": "Public Storage Account",
   "severity": "HIGH",
   "category": "Access Control",
-  "descriptionText": "Check if 'network_acls' is open to public.",
+  "descriptionText": "Storage Account should not be public",
```
Comment (rafaela-soares) [inline assets/queries/ansible/azure/public_storage_account/metadata.json:6]: This description could have more details about the impact. Do you feel comfortable trying to improve it?

**Review by rafaela-soares — CHANGES_REQUESTED** [review]
Hello, @roi-orca!
Hello, @roi-orca!

Our security team took the opportunity to analyze the queries of your PR. Some suggestions are not related to what you changed, but they will improve the queries.

Please, take a look at the suggestion and let me know what you think about it 😊 

Note that the suggestions apply to all platforms that have the query.

**Comment by roi-orca** [comment]
Hi @rafaela-soares,
Regarding the comments of `This description could have more details about the impact`
Do you mean how actually the issue should be resolved in the platform?

**Comment by rafaela-soares** [comment]
> Hi @rafaela-soares, Regarding the comments of `This description could have more details about the impact` Do you mean how actually the issue should be resolved in the platform?

Hi, @roi-orca! We mean how the vulnerability can impact the environment or the user. For example, regarding the query `DB Instance Publicly Accessible`, the fact that the db is public will make the attack surface bigger and more susceptible to attacks.

**Comment by gabriel-cx** [comment]
Hello, @roi-orca!
Feel free to ping me if you need something from our side in order to close this PR! ⛵
If you prefer, we can close it in our side.

**Review by cxMiguelSilva — APPROVED** [review]
LGTM

**Review by cxMiguelSilva — DISMISSED** [review]

**Review by cxMiguelSilva — DISMISSED** [review]

**Review by cxMiguelSilva — APPROVED** [review]

**Review by rafaela-soares — APPROVED** [review]

**Pull request merged**